gst_plugins/python/savant_rs_video_demux.py

```diff
@@ -424,11 +414,91 @@ def handle_video_frame(
                 'Pushing frame with idx=%s and pts=%s', frame_idx, frame_pts
             )
             result: Gst.FlowReturn = source_info.src_pad.push(frame_buf)
+
+        else:
+            result: Gst.FlowReturn = frame_buf
+
         self.logger.debug(
-            'end handle_buffer (return buffer with timestamp %d).', frame_pts
+            'Frame from source %s with PTS %s was processed.',
+            video_frame.source_id,
+            frame_pts,
         )
         return result
 
+    def _build_frame_buffer(
+        self,
+        video_frame: VideoFrame,
+        buffer: Gst.Buffer,
+    ) -> Union[Gst.Buffer, Gst.FlowReturn]:
```

Comment:
May be better to check content.is_none() out of function and to return only buffer or raise an exception, and wrap function call
```
try:
    frame_buf = self._build_frame_buffer(video_frame, buffer)
    ...
    return Gst.FlowReturn.OK
except ..:
    ...
    return Gst.FlowReturn.ERROR
```

Reply:
Wrapped in try/except.